Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -252,6 +276,20 @@ open class _ChatMessageListVC<ExtraData: ExtraDataTypes>:
     ) -> String? {
         overlayDateFormatter.string(from: channelController.messages[indexPath.item].createdAt)
     }
+
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let lastCellIsVisible = collectionView.isLastCellVisible
+        if lastCellWasVisible != lastCellIsVisible {
+            lastCellWasVisible = lastCellIsVisible
+
+            if lastCellIsVisible, channelController.channel?.isUnread == true {
+                channelController.markRead()
```

Comment:
Calling `markRead` multiple times is safe. LLC won't make an API call if not needed. So feel free to just call is as much as you want.